core/arch/arm/include/mm/core_mmu.h

```diff
@@ -86,6 +86,7 @@
  * MEM_AREA_IO_SEC:   Secure HW mapped registers
  * MEM_AREA_RES_VASPACE: Reserved virtual memory space
  * MEM_AREA_TA_VASPACE: TA va space, only used with phys_to_virt()
+ * MEM_AREA_TEE_RAM_UNCACHED:  Same as MEM_AREA_TEE_RAM, but cache disabled.
```

Comment:
If you plan, in the end, to push the whole suspend/resume work you did, maybe it could help if you push the whole as a RFC, or even not as a PR/RFC but only to your github account, for us to catch the full picture. In the meantime, you can push as specific PR some small pieces of the series so they can be easier review/matured/merged, as you did for this initial 3-patch series.